### PR TITLE
Fix #1054: Table Entity should have extended entity to capture SQL Queries

### LINKED ISSUE #1054: Table Entity should have extended entity to capture SQL Queries that ran against them

### PR DESCRIPTION
**Pull request by harshach**
### Describe your changes :
Display list of queries that are running against a table entity and allow users to vote up to recognize the popular queries.
Also register the duration of the query and user who ran them

#
### Type of change :

- [x] New feature

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have performed a self-review of my own. 
- [ ] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@sureshms

### CODE REVIEW

File: catalog-rest-service/src/main/resources/json/schema/entity/data/table.json [hunk 1]
```diff
@@ -343,6 +343,39 @@
...
       "additionalProperties": false
+    },
+    "sqlQuery": {
+      "type": "object",
+      "javaType": "org.openmetadata.catalog.type.SQLQuery",
+      "description": "This schema defines the type to capture the table's sql queries.",
+      "properties": {
+        "query": {
+          "description": "SQL Query text that matches the table name.",
+          "type": "string"
```
Comment (sureshms) [inline catalog-rest-service/src/main/resources/json/schema/entity/data/table.json:354]: there is sqlQuery in basic.json, right?

Comment (sureshms) [reply catalog-rest-service/src/main/resources/json/schema/entity/data/table.json:354]: maybe call this queryDetails?

File: catalog-rest-service/src/main/resources/json/schema/entity/data/table.json [hunk 2]
```diff
@@ -343,6 +343,39 @@
...
+        },
+        "duration": {
+          "description": "How long did the query took to run in seconds.",
+          "type": "number"
```
Comment (sureshms) [inline catalog-rest-service/src/main/resources/json/schema/entity/data/table.json:358]: would we capture sub seconds details? should we make this milli seconds?

**Review by sureshms — APPROVED** [review]

**Review by sureshms — COMMENTED** [review]
I see two abstractions here - `query `which sql statement with associated tables and an author. A `query run`, which is running of a query by a user that has duration etc. and might be run multiple times.

So `query` can be associated with views, metrics, reports etc. And might need to be an entity and `query run` is an execution of `query` with run details. This is a good start and we can work on getting the abstractions right over time keeping API backward compatibility. WDYT?

**Comment by harshach** [comment]
@sureshms Ya I thought about it. We can keep the query as a separate entity and link it to table , pipeline etc.. as you said. This can also be filtered down team, org level as well. "query run" could be the link to the pipeline itself? I understand that the query run , i.e duration can be different on each run.

**Pull request merged**